poetry_dynamic_versioning/__init__.py

```diff
@@ -668,7 +668,7 @@ def _get_and_apply_version(
     elif pep621:
         name = pyproject["project"]["name"]
         original = pyproject["tool"]["poetry"]["version"]
-        dynamic_array = pyproject["project"]["dynamic"].copy()
+        dynamic_array = pyproject["project"]["dynamic"]
```

Comment:
Turns out `.copy()` gets you the underlying `list`, not the `tomlkit.item.Array`.

Reply:
I wouldn't have guessed either 😅 Thanks for catching that.